src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -64,12 +67,19 @@ public async Task Invoke(HttpContext context)
                 return;
             }
 
-            // Policy evaluator has transient lifetime so it fetched from request services instead of injecting in constructor
+            // Policy evaluator has transient lifetime so it's fetched from request services instead of injecting in constructor
             var policyEvaluator = context.RequestServices.GetRequiredService<IPolicyEvaluator>();
 
             var authenticateResult = await policyEvaluator.AuthenticateAsync(policy, context);
 
-            // Allow Anonymous skips all authorization
+            if (authenticateResult.Succeeded)
```

Comment:
```suggestion
            if (authenticateResult?.Succeeded ?? false)
```

Reply:
AuthenticateAsync doesn't return a nullable result. Unless we just want to be extra protective in case a user returns null.

Reply:
IPolicyEvaluator is a user extensibility point, assume the worst 😁.